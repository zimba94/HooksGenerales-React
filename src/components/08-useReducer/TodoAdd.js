import React from 'react'
import { useForm } from '../../hooks/useForm';

export const TodoAdd = ({ handleAddTodo }) => {
    const [{description}, handleInputChange, reset] = useForm({
        description: ''
    });

    const handleSubmit = (e)=>{
        e.preventDefault();

        if (description.trim().length <= 1) {
            return;
        }
        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        };

        const action = {
            type: 'add',
            payload: newTodo
        };
        handleAddTodo(newTodo);
        reset();
    };

    return (
        <>
            <h4>Agregar ToDo</h4>
            <hr></hr>

            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    name="description"
                    className="form-control"
                    placeholder="Aprender ..."
                    autoComplete="off"
                    onChange={handleInputChange}
                    value={description}
                >
                </input>
                <button
                    type="submit"
                    className="btn btn-outline-primary mt-1 w-100"
                >
                    Agregar
                </button>
            </form>
        </>
    )
}
