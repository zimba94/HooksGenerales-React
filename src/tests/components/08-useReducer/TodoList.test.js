
import React from 'react';
import { shallow } from 'enzyme';
import { TodoList} from "../../../components/08-useReducer/TodoList";
import { demoTodos } from "../../fixtures/demoTodos";

describe('Pruebas en el componente TodoList', () => {
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow( 
        <TodoList
            todos={demoTodos}
            handleDelete={handleDelete}
            handleToggle={handleToggle}
        
        />);

    test('Debe mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener el mismo numero de todos que en demoTodos ', () => {
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);

        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
    });
});
