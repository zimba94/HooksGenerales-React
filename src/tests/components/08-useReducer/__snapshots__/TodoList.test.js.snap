// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente TodoList Debe mostrarse correctamente  1`] = `
<Fragment>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      i={0}
      key="1"
      todo={
        Object {
          "desc": "Aprender React",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      i={1}
      key="2"
      todo={
        Object {
          "desc": "Aprender Mongo",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</Fragment>
`;
